### PR TITLE
Add some basic settings + quickstart docs

### PR DESCRIPTION
**Pull request by i-trofimtschuk**

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -3,13 +3,44 @@
    You can adapt this file completely to your liking, but it should at least
    contain the root `toctree` directive.
 
-Welcome to redis-tasks's documentation!
+Welcome to redis-tasks' documentation!
 =======================================
 
+Redis tasks is a framework for background tasks execution.
+It uses redis for it's shared resources like queues.
```
Comment (vasilty) [inline docs/source/index.rst:10]:
```suggestion
It uses redis for its shared resources like queues.
```

File: docs/source/index.rst [hunk 2]
```diff
@@ -3,13 +3,44 @@
...
+It's able to provide reliable tasks execution for long
+running tasks even in environments, where it's expected
+that workers can be restarted at any time, like heroku.
```
Comment (vasilty) [inline docs/source/index.rst:13]:
```suggestion
that workers can be restarted at any time, like Heroku.
```

File: docs/source/index.rst [hunk 3]
```diff
@@ -3,13 +3,44 @@
...
+
+Disclaimer
+----------
+`redis-tasks` does not store any kind of task results.
```
Comment (vasilty) [inline docs/source/index.rst:17]: Why don't you use double '`'? Now you have italic and not monospace `redis-tasks`

File: docs/source/index.rst [hunk 4]
```diff
@@ -3,13 +3,44 @@
...
+using a persistence layer of your choice if needed.
+
+Installation
+============
+
+Your can install `redis-tasks` using pip like this::
+
+    $ pip install redis-tasks
+
+Or add `redis-tasks` to your requirements.in/.txt
```
Comment (vasilty) [inline docs/source/index.rst:28]: This suggests some magic. Just adding is to `requirements.txt` won't install. I would remove this line

File: docs/source/index.rst [hunk 5]
```diff
@@ -3,13 +3,44 @@
...
+
+You'll have to set the :envvar:`RT_SETTINGS_MODULE` or manually
```
Comment (vasilty) [inline docs/source/index.rst:30]:
```suggestion
You have to either set the :envvar:`RT_SETTINGS_MODULE` or manually
```

File: docs/source/settings.rst [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    Default for maximum number of seconds a task is allowed to be executed.
+    Can be overridden on a per task level using timeout argument of `@redis_task` decorator
+    Workhorse processes executing a task for longer than this amount of seconds
+    will be killed using 9/SIGKILL signal.
+
+.. attribute:: EXPIRING_REGISTRIES_TTL
+
+    :default: 604800  # 7 days
+
+    Registries are used for tracking successfully (`finished`) or unsuccessfully ('failed')
```
Comment (vasilty) [inline docs/source/settings.rst:42]: different handling of strings `finished` and `failed`

File: docs/source/settings.rst [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+.. attribute:: WORKER_DESCRIPTION_FUNCTION
+
+   :default: "redis_tasks.worker_process.generate_worker_description"
+
+.. attribute:: MIDDLEWARE
+
+    :default: []  # empty list
+
+    A list of middlewares to used, supplied as importable python dotted path strings
```
Comment (vasilty) [inline docs/source/settings.rst:71]:
```suggestion
    A list of task middlewares, supplied as importable python dotted path strings
```

File: docs/source/settings.rst [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+    `task`: Import path of the function to be run, e.g. "mymodule.myfunc"
+
+    `schedule`: Schedule for this task, e.g. crontab("2 4 * * mon,fri")
+
+    `args`, `kwargs`: args and kwargs for the tak function (optional)
+
+    `singleton`: Boolean to specify whether this task should not be on the queue multiple times (optional, defaults to True)
+
+    `queue`: Queue to put this task on (optional)
```
Comment (vasilty) [inline docs/source/settings.rst:91]: We should explain the missing value

File: docs/source/settings.rst [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+    `SCHEDULE` is the main configuration of tasks that should be run by the
+    the scheduler regularly.
+
+    It is an `id` -> `entry` dict.
+    The entries are dicts with the following keys:
+
+    `task`: Import path of the function to be run, e.g. "mymodule.myfunc"
+
+    `schedule`: Schedule for this task, e.g. crontab("2 4 * * mon,fri")
```
Comment (vasilty) [inline docs/source/settings.rst:85]: `mon, fri`? I wish it was that readable

**Comment by vasilty** [comment]
@i-trofimtschuk Shouldn't all settings be prefixed with `RT_`?

File: docs/source/index.rst [hunk 10]
```diff
@@ -3,13 +3,142 @@
...
+using a persistence layer of your choice if needed.
+
+Installation
+============
+
+Your can install `redis-tasks` using pip like this::
+
+    $ pip install redis-tasks
+
+Or add `redis-tasks` to your requirements.in/.txt
```
Comment (i-trofimtschuk) [inline docs/source/index.rst:28]:
```suggestion
```

File: docs/source/index.rst [hunk 11]
```diff
@@ -3,13 +3,140 @@
    You can adapt this file completely to your liking, but it should at least
    contain the root `toctree` directive.
 
-Welcome to redis-tasks's documentation!
+Welcome to redis-tasks' documentation!
 =======================================
 
+Redis tasks is a framework for background tasks execution.
```
Comment (kaloyanm) [inline docs/source/index.rst:9]:
```suggestion
 Redis tasks is a framework for background tasks execution.
 It is designed to provide reliable and long running tasks, even in environment 
where workers restarts quite often in cloud services like Heroku.
The task queueing is backed by Redis.
```

File: docs/source/index.rst [hunk 12]
```diff
@@ -3,13 +3,140 @@
...
+It uses redis for its shared resources like queues.
+It's able to provide reliable tasks execution for long
+running tasks even in environments, where it's expected
+that workers can be restarted at any time, like Heroku.
+
+Disclaimer
+----------
+`redis-tasks` does not store any kind of task results.
+It's your tasks' code responsibility to store task results
```
Comment (kaloyanm) [inline docs/source/index.rst:18]:
```suggestion
It's your job to write code responsible for dealing with it.
```

File: docs/source/index.rst [hunk 13]
```diff
@@ -3,13 +3,140 @@
...
+before you can use redis-tasks in your code.
+
+
+Quickstart Guide
+================
+
+Prerequisite: You have successfully installed redis-tasks and have a redis insteance ready.
+In this Quickstart Guide we'll use "redis://localhost:6379" as our REDIS_URL
+
+Very basic example
```
Comment (kaloyanm) [inline docs/source/index.rst:39]:
```suggestion
Jump start
```

Comment (kaloyanm) [reply docs/source/index.rst:39]: Or this title could be removed I think.

File: docs/source/index.rst [hunk 14]
```diff
@@ -3,13 +3,140 @@
...
+------------------
+1. Create a tasks module `tasks.py` and write a function that you want to execute in a worker using redis-tasks
```
Comment (kaloyanm) [inline docs/source/index.rst:41]:
```suggestion
1. Create `tasks.py` and write a function that you want to execute in a worker using redis-tasks
```

File: docs/source/index.rst [hunk 15]
```diff
@@ -3,13 +3,140 @@
...
+
+    import urllib.request
+    from redis_tasks import redis_task
+
+    @redis_task()
+    def print_len_of_url_content(url):
+        with urllib.request.urlopen(url) as f:
+            print(len(f.read()))
+
+2. Write an rt_settings module to supply your custom config to redis-tasks
```
Comment (kaloyanm) [inline docs/source/index.rst:53]:
```suggestion
2. Add `rt_settings.py` to supply your custom config to redis-tasks
```

File: docs/source/index.rst [hunk 16]
```diff
@@ -3,13 +3,140 @@
...
+    default_queue.enqueue_call(print_len_of_url_content,
+                               kwargs={'url': 'https://www.readthedocs.org/'})
+
+In the worker logs you should see the task being started and finished::
+
+    16:15:53 INFO      redis_tasks.task: Task tasks.print_len_of_url_content(url='https://www.google.com/') [095c3eb9-7073-4ad6-a7cc-16885ee962a4] started
+    12862
+    16:15:53 INFO      redis_tasks.task: Task tasks.print_len_of_url_content(url='https://www.google.com/') [095c3eb9-7073-4ad6-a7cc-16885ee962a4] finished
+
+Basic example (including scheduler)
```
Comment (kaloyanm) [inline docs/source/index.rst:90]:
```suggestion
With scheduling
```

File: docs/source/index.rst [hunk 17]
```diff
@@ -3,13 +3,140 @@
...
+-----------------------------------
+
+5. To add regularly executed tasks we extend the rt_settings module to this:
```
Comment (kaloyanm) [inline docs/source/index.rst:93]:
```suggestion
So far the configuration allowed you to run the tasks almost immediately. What if you want to specify the exact time?
5. First you should add to rt_settings module this:
```

File: docs/source/index.rst [hunk 18]
```diff
@@ -3,13 +3,140 @@
...
+
+Disclaimer
+----------
+`redis-tasks` does not store any kind of task results.
+It's your tasks' code responsibility to store task results
+using a persistence layer of your choice if needed.
+
+Installation
+============
+
```
Comment (dkusnik) [inline docs/source/index.rst:23]: Make sure you have python >=3.6 installed, then you can instal...

Comment (i-trofimtschuk) [reply docs/source/index.rst:23]: @dkusnik FYI we have it in setup.py
https://github.com/djangsters/redis-tasks/blob/master/setup.py#L25

but it's probably good to also mention it in the docs

File: docs/source/settings.rst [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+
+    The :envvar:`RT_SETTINGS_MODULE` environment variable can be used to override the 
+    default settings using a standard python module.
+    It is processed as a standard python dotted module path.
+
+.. attribute:: REDIS_URL
+
+    :default: "redis://127.0.0.1:6379"
+
+    A URL of the redis instance to be used by redis-tasks.
```
Comment (dkusnik) [inline docs/source/settings.rst:14]: An URL of the redis instance, which will be used by redis-tasks

File: docs/source/settings.rst (hunk 9 above)
Comment (dkusnik) [inline docs/source/settings.rst:85]: it is kind of hard to understand for me. Maybe an example, and depending on it the explanation?

File: docs/source/settings.rst [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+    the scheduler regularly.
+
+    It is an `id` -> `entry` dict.
+    The entries are dicts with the following keys:
+
+    `task`: Import path of the function to be run, e.g. "mymodule.myfunc"
+
+    `schedule`: Schedule for this task, e.g. crontab("2 4 * * mon,fri")
+
+    `args`, `kwargs`: args and kwargs for the tak function (optional)
```
Comment (dkusnik) [inline docs/source/settings.rst:87]:
```suggestion
    `args`, `kwargs`: args and kwargs for the task function (optional)
```

File: docs/source/settings.rst [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+    `queue`: Queue to put this task on (optional)
+
+.. attribute:: SCHEDULER_QUEUE
+
+    :default: "default"
+
+    Name of the queue to be used by the scheduler when enqueuing scheduled tasks.
+    It will be used when no `queue` key is provided on a SCHEDULE` entry.
+
+.. attribute:: SCHEDULER_MAX_CATCHUP
```
Comment (dkusnik) [inline docs/source/settings.rst:100]: what is this for?

File: docs/source/index.rst [hunk 22]
```diff
@@ -3,13 +3,140 @@
...
+   $ python -m redis_tasks.cli worker
+
+You should see a message that the worker started successfully::
+
+    16:15:39 INFO    redis_tasks.worker: Worker iw-T460p.14152 [0e015d7a-79c7-42be-8b02-add7161c5951] started
+
+4. Using an ipython shell or whereever needed in your code, you can
+   now enqueue this tasks to be executed by the worker at any time like this:
+
+.. note:: Make sure that your the python process is also started with the same ENV VARS as the worker process, see export calls in step 3.
```
Comment (groupsky) [inline docs/source/index.rst:74]:
```suggestion
.. note:: Make sure that the python process is also started with the same ENV VARS as the worker process, see export calls in step 3.
```

File: docs/source/index.rst (hunk 3 above)
Comment (i-trofimtschuk) [reply docs/source/index.rst:17]:
```suggestion
``redis-tasks`` does not store any kind of task results.
```

File: docs/source/index.rst [hunk 23]
```diff
@@ -3,13 +3,140 @@
...
+It's your tasks' code responsibility to store task results
+using a persistence layer of your choice if needed.
+
+Installation
+============
+
+Your can install `redis-tasks` using pip like this::
+
+    $ pip install redis-tasks
+
```
Comment (i-trofimtschuk) [inline docs/source/index.rst:27]:
```suggestion

.. note:: ``redis-tasks`` only runs on python versions 3.6 and above.

```

File: docs/source/index.rst (hunk 11 above)
Comment (bastir85) [reply docs/source/index.rst:9]: i like this more, i stunpled upon this sentence as well.

File: docs/source/index.rst [hunk 24]
```diff
@@ -3,13 +3,142 @@
...
 
-Welcome to redis-tasks's documentation!
+Welcome to redis-tasks' documentation!
 =======================================
 
+Redis tasks is a framework for background tasks execution.
+It uses redis for its shared resources like queues.
+It's able to provide reliable tasks execution for long
+running tasks even in environments, where it's expected
+that workers can be restarted at any time, like Heroku.
```
Comment (i-trofimtschuk) [inline docs/source/index.rst:13]:
```suggestion
Redis tasks is a framework for background tasks execution.
Redis tasks is a framework for background tasks execution.
 It is designed to provide reliable and long running tasks, even in environment 
where workers restarts quite often on cloud services like Heroku.
The task queueing is backed by Redis.
```

Comment (dkusnik) [reply docs/source/index.rst:13]: Redis-tasks is a framework for executing and maintaining background tasks. It is able to provide reliability in environments where workers often restart, like in cloud environments (e.g. heroku). This allows running time consuming tasks without concerns about the proper execution. The task queuing is based on Redis.

File: docs/source/index.rst [hunk 25]
```diff
@@ -3,13 +3,142 @@
...
+
+Disclaimer
+----------
+``redis-tasks`` does not store any kind of task results.
+It's your tasks' code responsibility to store task results
+using a persistence layer of your choice if needed.
```
Comment (i-trofimtschuk) [inline docs/source/index.rst:19]:
```suggestion
If you need to store tasks results, you have to use
a persistence layer of your choice in your task code.
```

File: docs/source/index.rst [hunk 26]
```diff
@@ -3,13 +3,142 @@
...
+
+You have to either set the :envvar:`RT_SETTINGS_MODULE` or manually
+call `settings.configure()` passing in a python settings module
+before you can use redis-tasks in your code.
+
+
+Quickstart Guide
+================
+
+Prerequisite: You have successfully installed redis-tasks and have a redis insteance ready.
```
Comment (i-trofimtschuk) [inline docs/source/index.rst:38]:
```suggestion
Prerequisite: You have successfully installed redis-tasks and have a running redis instance.
```

File: docs/source/index.rst (hunk 15 above)
Comment (i-trofimtschuk) [reply docs/source/index.rst:55]: unify with 1.

File: docs/source/index.rst [hunk 27]
```diff
@@ -3,13 +3,142 @@
...
+
+   $ export REDIS_URL='redis://127.0.0.1:6379'
+   $ export RT_SETTINGS_MODULE=rt_settings 
+   $ python -m redis_tasks.cli worker
+
+You should see a message that the worker started successfully::
+
+    16:15:39 INFO    redis_tasks.worker: Worker iw-T460p.14152 [0e015d7a-79c7-42be-8b02-add7161c5951] started
+
+4. Using an ipython shell or whereever needed in your code, you can
```
Comment (i-trofimtschuk) [inline docs/source/index.rst:73]: * In another separate terminal/shell 
* remove whereever in your code part

File: docs/source/index.rst [hunk 28]
```diff
@@ -3,13 +3,142 @@
...
+            'schedule': run_every(minutes=20),
+            #'schedule': once_per_day('06:00'),  # Alternative 1
+            #"schedule": crontab("0 0 * * 1")},  # Alternative 2
+            'args': [],
+            'kwargs': {'url': 'https://readthedocs.org/'},
+        },
+    }
+
+
+6. In a separate shell/terminal Start a scheduler process::
```
Comment (i-trofimtschuk) [inline docs/source/index.rst:117]:
```suggestion
6. In a separate shell/terminal start a scheduler process::
```

File: docs/source/index.rst [hunk 29]
```diff
@@ -3,13 +3,142 @@
...
+
+    REDIS_URL = os.environ['REDIS_URL']
+
+    TIMEZONE = "UTC"  # timezone to be used by scheduler
+    SCHEDULE = {
+        'print_len_of_readthedocs_org': {  # the task entry identifier
+            'task': 'tasks.print_len_of_url_content',  # dotted module path to function to be executed
+            'schedule': run_every(minutes=20),
+            #'schedule': once_per_day('06:00'),  # Alternative 1
+            #"schedule": crontab("0 0 * * 1")},  # Alternative 2
```
Comment (i-trofimtschuk) [inline docs/source/index.rst:110]:
```suggestion
```

File: docs/source/index.rst [hunk 30]
```diff
@@ -3,13 +3,142 @@
...
+
+Basic example (including scheduler)
+-----------------------------------
+
+5. To add regularly executed tasks we extend the rt_settings module to this:
+
+.. code:: python
+
+    import os
+    from redis_tasks import crontab, once_per_day, run_every
```
Comment (i-trofimtschuk) [inline docs/source/index.rst:100]:
```suggestion
    from redis_tasks import run_every
```

File: docs/source/settings.rst (hunk 19 above)
Comment (i-trofimtschuk) [inline docs/source/settings.rst:14]:
```suggestion
    The URL of the redis instance to be used by redis-tasks.
```

File: docs/source/settings.rst [hunk 31]
```diff
@@ -0,0 +1,102 @@
...
+
+    :default: "redis://127.0.0.1:6379"
+
+    A URL of the redis instance to be used by redis-tasks.
+
+.. attribute:: REDIS_PREFIX
+
+    :default: "redis_tasks"
+
+    A redis key prefix, that will be used as a namespace for all redis keys used by redis-tasks.
```
Comment (i-trofimtschuk) [inline docs/source/settings.rst:20]:
```suggestion
    The redis key prefix, that will be used as a namespace for all redis keys used by redis-tasks.
```

File: docs/source/settings.rst (hunk 9 above)
Comment (i-trofimtschuk) [reply docs/source/settings.rst:85]: add example

File: docs/source/settings.rst (hunk 8 above)
Comment (i-trofimtschuk) [reply docs/source/settings.rst:91]: default

**Pull request closed without merging**